prow/git/git.go

```diff
@@ -601,3 +633,18 @@ func (r *Repo) Fetch(arg ...string) error {
 	}
 	return nil
 }
+
+func buildMergeRebaseArgs(commitlike string, opts ...MergeOpt) ([]string, error) {
+	if commitlike == "" {
+		return nil, errors.New("branch must be set")
+	}
+	args := []string{"rebase", "--no-stat"}
+	if len(opts) == 0 {
+		return append(args, commitlike), nil
+	}
+	opt := opts[0]
```

Comment:
If you use variadic arg opts you have to look at all of them, not only at the first.

More importantly, why are we going through the hassle of passing the `UpstreamBranchSha` around? Apart from making the method super inconveninent to use and easy to misuse it isn't needed, because the git client is on the correct branch already, so you can find it out through `git rev-parse HEAD`

Reply:
You are correct, thank you for the explanation. I missed [this one](https://github.com/kubernetes/test-infra/blob/master/prow/git/v2/interactor.go#L51).